prov/gni/src/gnix_cm_nic.c

```diff
@@ -476,14 +474,13 @@ int _gnix_cm_nic_free(struct gnix_cm_nic *cm_nic)
 
 int _gnix_cm_nic_alloc(struct gnix_fid_domain *domain,
 		       struct fi_info *info,
-		       struct gnix_cm_nic **cm_nic_ptr)
+		       struct gnix_cm_nic **cm_nic_ptr,
+		       uint32_t name_type,
+		       uint32_t cdm_id)
```

Comment:
would you mind rearranging the arguments so that ```struct gnix_cm_nic **cm_nic_ptr``` is the last one?  
I think we may be able to drop the ```name_type``` argument.